README.md

```diff
@@ -2,54 +2,113 @@
+BMC information for a discovered node.
+
+Destruction of a node object removes the node from Ironic.
+
+
+```
 resource "ironic_node_v1" "openshift-master-0" {
   name = "openshift-master-0"
-  target_provision_state = "active"
-  user_data = "${file("master.ign")}"
 
   ports = [
     {
-      "address" = "00:bb:4a:d0:5e:38"
-      "pxe_enabled" = "true"
-    }
+      pxe_enabled = true
+      address     = "00:bb:4a:d0:5e:38"
+    },
   ]
-
-  properties {
-    "local_gb" = "50"
-    "cpu_arch" =  "x86_64"
-  }
-
-  instance_info = {
-    "image_source" = "http://172.22.0.1/images/redhat-coreos-maipo-latest.qcow2"
-    "image_checksum" = "26c53f3beca4e0b02e09d335257826fd"
-    "root_gb" = "25"
-    "root_device" = "/dev/vda"
-  }
-
+  
   driver = "ipmi"
+  
   driver_info {
-			"ipmi_port"=      "6230"
-			"ipmi_username"=  "admin"
-			"ipmi_password"=  "password"
-			"ipmi_address"=   "192.168.111.1"
-			"deploy_kernel"=  "http://172.22.0.1/images/ironic-python-agent.kernel"
-			"deploy_ramdisk"= "http://172.22.0.1/images/ironic-python-agent.initramfs"
+    ipmi_port       =  "6230"
+    ipmi_username   =  "admin"
+    ipmi_password   =  "password"
+    ipmi_address    =  "192.168.111.1"
+    deploy_kernel   =  "http://172.22.0.1/images/ironic-python-agent.kernel"
+    deploy_ramdisk  =  "http://172.22.0.1/images/ironic-python-agent.initramfs"
   }
 }
 ```
 
+## Ports
+
+Ports may be specified as part of the node resource, or as a separate `ironic_port_v1`
+declaration.
+
+
+```
+resource "ironic_port_v1" "openshift-master-0-port-0" {
+  node_uuid   = "${ironic_node_v1.openshift-master-0.id}"
+  pxe_enabled = true
+  address     = "00:bb:4a:d0:5e:38"
+}
+```
+
+## Deployment
+
+A deployment will provision a baremetal node, using the information given in the resource.  The `count` metaparameter
+can be used to deploy multiple nodes.
+
+An available node will be selected based on critera described below, and terraform will drive
+the [Ironic state machine](https://docs.openstack.org/ironic/latest/contributor/states.html) to bring the node to an `active` state.
+
+Destruction of a deployment cleans a node and brings it back to the `available` state.
+
+### Node selection
+
+Users can control node selection with two fields. `candidates` is a list of node UUID's that
+could be used.  If specified, the first available node will be used.  The other mechanism is
+by specifying a list of `capabilities` (TODO: provide example), which will restrict which nodes are considered.
+ 
+If neither candidates nor capabilities are specified, the first available node will be used.
+
+```
+resource "ironic_deployment" "masters"  {
+  count = 3
+  
+  candidates = [
+    "${ironic_node_v1.openshift-master-0.id}",
+    "${ironic_node_v1.openshift-master-1.id}",
+    "${ironic_node_v1.openshift-master-2.id}",
+  ]
+
+  local_gb       = "50"
+  cpu_arch       =  "x86_64"
+  root_gb        = "25"
+  root_device    = "/dev/vda"
```

Comment:
Yes - I'll change this to be a map to reflect the various different [options](https://docs.openstack.org/ironic/pike/install/include/root-device-hints.html), so this would be:

```
root_device {
  name = "/dev/vda"
}
```